models/intermediate/int_klaviyo__event_attribution.sql

```diff
@@ -2,10 +2,7 @@
     config(
         materialized='incremental',
         unique_key='unique_event_id',
-        partition_by={
-            "field": "occurred_on",
-            "data_type": "date"
-        } if target.type == 'bigquery' else none,
+        partition_by={"field": "occurred_on", "data_type": "date", 'granularity': 'month'} if target.type == 'bigquery' else none,
```

Comment:
Unlike other packages, we are partitioning by a field that isn't used in any filters or joins. I am thinking for this fix, it might make more sense just to remove the partition_by to simplify the model. I did a quick test with our test data of a about 17 rows, and after removing the partitions, both full refresh and incremental runs were slightly faster by a few seconds.

The other option would be to update the field we partition on to what is used in the incremental filter, but I think that is out of scope. Later if we get around to overhauling the incremental logic anyway, we can add a partition back if necessary.

Reply:
> The other option would be to update the field we partition on to what is used in the incremental filter, but I think that is out of scope. Later if we get around to overhauling the incremental logic anyway, we can add a partition back if necessary.

Good callout! I say we go the route of removing the partition since it isn't doing anything to help and causing more issues than benefits. However, we can create a FR to explore adding a partition statement if we see requests for this type of enhancement and we can spend dedicated time add the proper partitioning logic.

I'll update the PR to remove the partition and adjust the documentation accordingly. I'll let you know once those are applied and this is ready for re-review.

Reply:
[FR](https://github.com/fivetran/dbt_klaviyo/issues/42) created to log this for a possible future effort if there is interest/need.